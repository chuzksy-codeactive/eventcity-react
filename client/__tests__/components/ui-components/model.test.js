import React from 'react'
import Connected, { Modal } from '../../../src/components/ui-components/Modal'

describe('Connected Component', () => {
  it('should render the connected component with all props', () => {
    const store = mockStore({
      centerReducer: {
        center: {
          message: 'message'
        }
      }
    })

    const wrapper = shallow(<Connected store={store} />);
    expect(wrapper.length).toEqual(1);
  });
});

describe('Modal Component test', () => {
  const props = {
    message: 'message'
  }
  const wrapper = shallow(<Modal {...props}/>)
  it('It should render the component without error', () => {
    expect(wrapper).toMatchSnapshot();
  });
})