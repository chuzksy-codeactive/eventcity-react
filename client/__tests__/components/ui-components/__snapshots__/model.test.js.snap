// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component test It should render the component without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    message="message"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modal-content"
      >
        <span
          className="close"
        >
          ×
        </span>
        <p>
          Some text in the Modal
        </p>
      </div>,
      "className": "myModal",
      "id": "modal",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="close"
          >
            ×
          </span>,
          <p>
            Some text in the Modal
          </p>,
        ],
        "className": "modal-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "×",
            "className": "close",
          },
          "ref": null,
          "rendered": "×",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some text in the Modal",
          },
          "ref": null,
          "rendered": "Some text in the Modal",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal-content"
        >
          <span
            className="close"
          >
            ×
          </span>
          <p>
            Some text in the Modal
          </p>
        </div>,
        "className": "myModal",
        "id": "modal",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="close"
            >
              ×
            </span>,
            <p>
              Some text in the Modal
            </p>,
          ],
          "className": "modal-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "×",
              "className": "close",
            },
            "ref": null,
            "rendered": "×",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Some text in the Modal",
            },
            "ref": null,
            "rendered": "Some text in the Modal",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
